command-line/lib/Options/Applicative/Script.hs

```diff
@@ -0,0 +1,133 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+{-# OPTIONS_HADDOCK hide #-}
+
+module Options.Applicative.Script
+    (
+    -- ** Applicative Parser
+      scriptArg
+
+    -- * Internal
+    , requireSignatureOfParser
+    , requireAllOfParser
+    , requireAnyOfParser
+    , requireAtLeastOfParser
+    , scriptParser
+
+    ) where
+
+import Prelude
+
+import Cardano.Multisig
+    ( Script (..), keyHashFromBytes, validateScript )
+import Codec.Binary.Encoding
+    ( fromBase16 )
+import Data.Char
+    ( isDigit, isLetter )
+import Data.Word
+    ( Word8 )
+import Options.Applicative
+    ( Parser, argument, eitherReader, help, metavar )
+import Text.ParserCombinators.ReadP
+    ( ReadP, readP_to_S, (<++) )
+
+import qualified Data.Text as T
+import qualified Data.Text.Encoding as T
+import qualified Text.ParserCombinators.ReadP as P
+
+
+scriptFromString :: String -> Either String Script
```

Comment:
Maybe the error type could be `data ScriptError = MalformedScript | InvalidScript`

Reply:
I will deal with this(or at least try)  in the forthcoming PR